src/reachy2_sdk/parts/arm.py

```diff
@@ -394,11 +394,28 @@ def _goto_cartesian_interpolation(
             request = ArmCartesianGoal(
                 id=self._part_id,
                 goal_pose=Matrix4x4(data=interpolated_matrix.flatten().tolist()),
-                duration=FloatValue(value=time_step),
             )
             self._arm_stub.SendArmCartesianGoal(request)
             time.sleep(time_step)
 
+        current_pose = self.forward_kinematics()
+        precision_distance_xyz = np.linalg.norm(current_pose[:3, 3] - target[:3, 3])
+        if precision_distance_xyz > 0.003:
```

Comment:
This is a magic number. e.g it could be a parameter in the function with 0.003 as default.